Package.swift

```diff
@@ -16,7 +16,8 @@ var products: [Product] = [
     .library(name: "Sentry-WithoutUIKitOrAppKit", targets: ["Sentry-WithoutUIKitOrAppKit", "SentryCppHelper"]),
     .library(name: "Sentry-WithoutUIKitOrAppKit-WithARM64e", targets: ["Sentry-WithoutUIKitOrAppKit-WithARM64e", "SentryCppHelper"]),
     .library(name: "SentrySwiftUI", targets: ["Sentry", "SentrySwiftUI", "SentryCppHelper"]),
-    .library(name: "SentryDistribution", targets: ["SentryDistribution"])
+    .library(name: "SentryDistribution", targets: ["SentryDistribution"]),
+    .library(name: "SentrySwiftLog", targets: ["Sentry", "SentrySwiftLog"])
```

Comment:
`m`: I don't fully understand why this is another target. Is this only here because `swift-log` is a new dependency? Shouldn't we add it as a dependency to Sentry instead?

Reply:
Ok, there is a proposal that was partially adopted:

https://github.com/swiftlang/swift-evolution/blob/main/proposals/0226-package-manager-target-based-dep-resolution.md

I created a sample app with a similar setup. While SPM will always resolve the dependency, it will bot be linked/build if it is not required by the target.

So this setup works as expected.